src/Revision/EnhancedEntityRevisionTrait.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\entity\Revision\EnhancedEntityRevisionTrait.
+ */
+
+namespace Drupal\entity\Revision;
+use Drupal\user\Entity\User;
+
+/**
+ * Provides a trait implementing \Drupal\entity\EnhanceredEntityRevisionInterface.
+ */
+trait EnhancedEntityRevisionTrait {
+
+  /**
+   * Returns whether the entity type has a specific field.
+   *
+   * @param string $field_name
+   *   The field name.
+   *
+   * @return bool
+   */
+  abstract function hasField($field_name);
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getRevisionCreationTime() {
+    if ($this->hasField('revision_create')) {
```

Comment:
Let's jsut go with the field names without if any provide the fields in some helper method, based upon a commo naming pattern?
Like 
TRAITdefaultBaseFields() ?

That really starts to go into the direction of behaviours as traits..